transport/pool.go

```diff
@@ -80,7 +80,11 @@ func (p *ConnPool) LeastBusyConn() *Conn {
 			}
 		}
 	}
-	return leastBusyConn
+
+	if leastBusyConn == nil {
+		return nil, fmt.Errorf("no connections available for host %s", p.host)
```

Comment:
Do I understand correctly that, when the node goes down, the pool becomes empty and this error will be returned a lot? If so, then maybe we could also have a global instance for this error?

Reply:
I think this could depend on how #293 is handled, for example if pool losing all connections would automatically mark the node down, this error would only be called in the time between the last conn dying and status being marked.

I'd keep it as simple as it is right now.